src/lsqecc/pauli_rotations/rotation.py

```diff
@@ -366,6 +368,29 @@ def from_cz_gate(
 
         return [entangling_op, correct_control, correct_target]
 
+    @staticmethod
+    def from_crz_gate(
+        num_qubits: int, control_qubit: int, target_qubit: int, phase: Fraction
+    ) -> List["PauliRotation"]:
+        """
+        Use the follwing identity:
+        q_0: ─────■─────               ┌───────────┐
+             ┌────┴────┐    ---   q_0: ┤ Rz(π/(2n))├──■──────────────────■──
+        q_1: ┤ Rz(π/n) ├    ---        ├───────────┤┌─┴─┐┌────────────┐┌─┴─┐
+             └─────────┘          q_1: ┤ Rz(π/(2n))├┤ X ├┤ Rz(-π/(2n))├┤ X ├
+                                       └───────────┘└───┘└────────────┘└───┘
+        """
+        Z = PauliOperator.Z
+
+        return (
+            [
+                PauliRotation.from_r_gate(num_qubits, control_qubit, Z, phase / 2),
+                PauliRotation.from_r_gate(num_qubits, target_qubit, Z, phase / 2),
+            ]
+            + PauliRotation.from_cnot_gate(num_qubits, control_qubit, target_qubit)
+            + [PauliRotation.from_r_gate(num_qubits, target_qubit, Z, -phase / 2)]
+        )
```

Comment:
@isolatedinformation, the identity you showed me

Reply:
https://www.nature.com/articles/s41598-018-23764-x.pdf?origin=ppub, just putting the paper here, cus I keep losing it

Reply:
@isolatedinformation Of course, but George did not trust my circuit identity skills :)